src/Html2Text.php

```diff
@@ -295,19 +316,35 @@ static function iterateOverNode($node, $prevName = null, $in_pre = false, $is_of
 					|| $n instanceof \DOMProcessingInstruction
 					|| ($n instanceof \DOMText && static::isWhitespace($text))) {
 					// Keep current previousSiblingName, these are invisible
+					$trailing_whitespace++;
 				}
 				else {
 					$previousSiblingName = strtolower($n->nodeName);
+					$previousSiblingNames[] = $previousSiblingName;
+					$trailing_whitespace = 0;
 				}
 
 				$node->removeChild($n);
 				$n = $node->childNodes->item(0);
 
-				// suppress last br tag inside a node list
-				if ($n != null || $previousSiblingName != 'br') {
-					$output .= $text;
+				$parts[] = $text;
+			}
+
+			// Remove trailing whitespace, important for the br check below
+			while($trailing_whitespace-- !== 0) {
```

Comment:
I would rather have `while ($trailing_whitespace-- > 0)` to prevent any possible infinite loop.